From b66c4c3f2ba2ab86eb78b9417246ae3de15b215b Mon Sep 17 00:00:00 2001
From: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
Date: Wed, 9 Oct 2024 09:52:16 +0530
Subject: [PATCH] mmc: card: enable partition scan for eMMC boot areas

This manually reverts the commit f5b4d71f72c5 ("mmc: card: Prevent
partition scan for the eMMC boot areas").

In Qualcomm IPQ5424 SoC, Xtensible Boot Loader(XBL) will be part of the
BOOT0 partition. Currently XBL partition information is not populated
under /sys/block/mmcblk0boot0. So enable the partition scan for BOOT
areas as well.

After enabling the partition scan, able to see the required partition
information.

[    1.802504] mmcblk0boot0: mmc0:0001 008GB1 4.00 MiB
[    1.821197]  mmcblk0boot0: p1 p2

root@OpenWrt:/# grep -Rn PARTNAME /sys/block/mmcblk0boot0/mmcblk0boot0p1/uevent
7:PARTNAME=0:XBL
root@OpenWrt:/# grep -Rn PARTNAME /sys/block/mmcblk0boot0/mmcblk0boot0p2/uevent
7:PARTNAME=0:XBL_1
root@OpenWrt:/#

Change-Id: Id3b7403967d8e937ce91fc4337461efec27c9cfb
Signed-off-by: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
---
 drivers/mmc/core/block.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/core/block.c b/drivers/mmc/core/block.c
index 152dfe593c43..cde65efc1b4f 100644
--- a/drivers/mmc/core/block.c
+++ b/drivers/mmc/core/block.c
@@ -2473,7 +2473,7 @@ static struct mmc_blk_data *mmc_blk_alloc_req(struct mmc_card *card,
 	md->disk->private_data = md;
 	md->parent = parent;
 	set_disk_ro(md->disk, md->read_only || default_ro);
-	if (area_type & (MMC_BLK_DATA_AREA_RPMB | MMC_BLK_DATA_AREA_BOOT))
+	if (area_type & MMC_BLK_DATA_AREA_RPMB)
 		md->disk->flags |= GENHD_FL_NO_PART;
 
 	/*
-- 
2.34.1

